src/test/java/uk/gov/companieshouse/registeredemailaddressapi/unit/controller/EligibilityControllerTest.java

```diff
@@ -0,0 +1,84 @@
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.ResponseEntity;
+
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.registeredemailaddressapi.controller.EligibilityController;
+import uk.gov.companieshouse.registeredemailaddressapi.eligibility.EligibilityStatusCode;
+import uk.gov.companieshouse.registeredemailaddressapi.exception.CompanyNotFoundException;
+import uk.gov.companieshouse.registeredemailaddressapi.exception.ServiceException;
+import uk.gov.companieshouse.registeredemailaddressapi.model.response.CompanyValidationResponse;
+import uk.gov.companieshouse.registeredemailaddressapi.service.CompanyProfileService;
+import uk.gov.companieshouse.registeredemailaddressapi.service.EligibilityService;
+
+@ExtendWith(MockitoExtension.class)
+class EligibilityControllerTest {
+
+    private static final String COMPANY_NUMBER = "11111111";
+    private static final String ERIC_REQUEST_ID = "XaBcDeF12345";
+
+    @Mock
+    private CompanyProfileService companyProfileService;
+
+    @Mock
+    private EligibilityService eligibilityService;
+
+    @InjectMocks
+    private EligibilityController eligibilityController;
+
+    @Test
+    void testSuccessfulGetEligibility() throws ServiceException, CompanyNotFoundException {
+        CompanyProfileApi companyProfileApi = new CompanyProfileApi();
+        companyProfileApi.setCompanyStatus("AcceptValue");
+        when(companyProfileService.getCompanyProfile(COMPANY_NUMBER)).thenReturn(companyProfileApi);
+        CompanyValidationResponse companyValidationResponse = new CompanyValidationResponse();
+        companyValidationResponse.setEligibilityStatusCode(EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE);
+        when(eligibilityService.checkCompanyEligibility(companyProfileApi)).thenReturn(companyValidationResponse);
+        ResponseEntity<CompanyValidationResponse> response = eligibilityController.getEligibility(COMPANY_NUMBER, ERIC_REQUEST_ID);
+        assertEquals(200, response.getStatusCodeValue());
+    }
+
+    @Test
+    void testFailedGetEligibility() throws ServiceException, CompanyNotFoundException {
+        CompanyProfileApi companyProfileApi = new CompanyProfileApi();
+        companyProfileApi.setCompanyStatus("FailureValue");
+        when(companyProfileService.getCompanyProfile(COMPANY_NUMBER)).thenReturn(companyProfileApi);
+        CompanyValidationResponse companyValidationResponse = new CompanyValidationResponse();
+        companyValidationResponse.setEligibilityStatusCode(EligibilityStatusCode.INVALID_COMPANY_STATUS);
+        when(eligibilityService.checkCompanyEligibility(companyProfileApi)).thenReturn(companyValidationResponse);
+        ResponseEntity<CompanyValidationResponse> response = eligibilityController.getEligibility(COMPANY_NUMBER, ERIC_REQUEST_ID);
+        assertEquals(200, response.getStatusCodeValue());
```

Comment:
This test has the same assertion (status code 200) as the previous happy path one - is there a further assertion we could add to each to differentiate their behaviour?

Reply:
Good spot, in a hard to read test (a bit of a lazy CAP from confirmation-statement-api).  Now also tidied-up for improved readability I hope!